src/datachain/query/batch.py

```diff
@@ -28,9 +36,9 @@ class BatchingStrategy(ABC):
     @abstractmethod
     def __call__(
         self,
-        execute: Callable,
-        query: sa.sql.selectable.Select,
-    ) -> Generator[BatchingResult, None, None]:
+        execute: Callable[..., Generator[Sequence, None, None]],
+        query: "Select",
+    ) -> Generator[RowsInput, None, None]:
```

Comment:
Q: should the name be RowsOutput or RowsResult?

Reply:
Sounds good to me, let me recheck. Naming is hard 👀